csharp/Adapter/Microsoft.Spark.CSharp/Configuration/ConfigurationService.cs

```diff
@@ -150,14 +150,25 @@ internal SparkCLRLocalConfiguration(System.Configuration.Configuration configura
 
             internal override string GetCSharpRDDExternalProcessName()
             {
-                return appSettings.Settings["CSharpWorkerPath"].Value;
+                KeyValueConfigurationElement workerPathConfig = appSettings.Settings["CSharpWorkerPath"];
+                if (workerPathConfig != null)
+                {
+                    // Use the file name part of the CSharpWorkerPath value, if specified
+                    return Path.GetFileName(workerPathConfig.Value);
```

Comment:
Actually, the more I look at this, seems that GetCSharpWorkerPath method is never used, so maybe should just rename GetCSharpRDDExternalProcessName?